test/test_security_service.py

```diff
@@ -103,7 +99,7 @@ def test_crud_security_group_service(self):
         sg = self.provider.security.security_groups.create(
             name=name, description=name)
         #Empty security groups don't exist in GCE. Let's add a dummy rule.
```

Comment:
@chiniforooshan @mbookman @afgane I think that we need to capture this constraint at the interface level and remove the line which manually adds a dummy rule in the test. Right now, the GCE behaviour is different from the AWS and OS implementations. Calling SecurityGroup.Create() results in a new resource being created in the latter, but a dummy wrapper being returned in the former. An actual security group will only be created if you add at least one rule in GCE. This means that a user who calls create() and expects the group to exist, will get unexpected behaviour in GCE.

One solution could be to change the create method's signature at the interface level to create(sg_name, initialRules, description=none), where initialRules is a non-null set of rules that can be specified when creating the securityGroup. This ensures that it's impossible, by design, to create a security group without at least a single rule.

We could also consider radically different designs. My understanding of GCE  rules though is that, rules (a.k.a firewalls) exist in isolation, and can be tagged. These tags can subsequently be assigned to multiple instances, in effect applying the rule. Ehsan's implementation groups these rules by tag to simulate a SecurityGroup. Is that correct? If you remove a rule, only the tag is removed? What happens when all tags are removed, is the rule automatically deleted by GCE?